### PR TITLE
Activates subscription through webhook

### PR DESCRIPTION
**Pull request by felipefreitag**

### CODE REVIEW

File: app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -27,6 +28,16 @@ def show
     authorize @subscription
   end
 
+  def activate
+    subscription = Subscription.find_by!(iugu_id: params[:data][:id])
+    authorize subscription
+    subscription.activate!
+    subscription.update!(
+      state: subscription.state,
```
Comment (danielweinmann) [inline app/controllers/subscriptions_controller.rb]: Our state machine already updated the state, so you don't need to update it here.

File: app/controllers/subscriptions_controller.rb [hunk 2]
```diff
@@ -27,6 +28,16 @@ def show
...
+      activated_at: subscription.activated_at
```
Comment (danielweinmann) [inline app/controllers/subscriptions_controller.rb]: [This line](https://github.com/felipefreitag/pangeas/blob/master/app/models/subscription.rb#L33) already sets `activated_at` when you `activate!`, so you don't need this update as well :)

File: app/controllers/subscriptions_controller.rb (hunk 1 above)
Comment (felipefreitag) [reply app/controllers/subscriptions_controller.rb]: When I tested, the `activate` was changing the object in memory, but it wasn't saving the changes to the DB. That's why I manually added the `update`. Is there another way to do this?

Comment (felipefreitag) [reply app/controllers/subscriptions_controller.rb]: Nevermind, I had something else wrong when I first tested it, now I removed the update and it worked :)

**Pull request merged**